session/session.go

```diff
@@ -3521,7 +3525,10 @@ func createSessionWithOpt(store kv.Storage, opt *Opt) (*session, error) {
 // We need this because when we start DDL in Domain, the DDL need a session
 // to change some system tables. But at that time, we have been already in
 // a lock context, which cause we can't call createSession directly.
-func CreateSessionWithDomain(store kv.Storage, dom *domain.Domain) (*session, error) {
+// attachStatsCollector indicates whether attach the stats collector on the
+// dom to the created session. The DDL session is not attached with a stats
+// collector, while others (e.g. TTL) are.
+func CreateSessionWithDomain(store kv.Storage, dom *domain.Domain, attachStatsCollector bool) (*session, error) {
```

Comment:
Attaching the collector to ddl will make a lot of tests (about analyze) flaky, so I add an option to avoid attaching the collector to the session excepts ttl.

Reply:
From what I am reviewed, I think API can be better:

1. Use`attachStatsCollector(*session) *session` to avoid API changes
2. Use `opts ... func(*session)` to make it more flexible